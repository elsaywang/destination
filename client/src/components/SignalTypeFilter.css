:local(.heading) {
    display: block;
    padding-left: 29px;
    font-size: 1.2em;
    font-weight: lighter;
}

:local(.signalType) {
    padding: 0 20px;
}
