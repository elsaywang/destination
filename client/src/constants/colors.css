/* TODO: utilize https://git.corp.adobe.com/Spectrum/spectrum-dna for css vars */

@value borderGrey: #CCC;
