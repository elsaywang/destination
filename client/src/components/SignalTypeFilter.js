import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { SideNav, SideNavItem, SideNavHeading } from '@react/react-spectrum/SideNav';
import Heading from '@react/react-spectrum/Heading';
import styles from './SignalTypeFilter.css';
import { signalTypeOptions, categoryTypes } from '../constants/signalTypeOptions';

class SignalTypeFilter extends Component {
    state = {
        value: 'ALL',
    };

    componentDidUpdate(prevProps) {
        if (!this.props.isSearched && prevProps.isSearched) {
            this.setState({ value: 'ALL' });
        }
    }

    handleSignalTypeSelect = value => {
        this.setState({ value }, () => this.props.onSignalTypeChange(value));
    };

    render() {
        return (
            <Fragment>
                <Heading size={3} className={styles.heading}>
                    Filter by Signal Type
                </Heading>
                <SideNav
                    autoFocus
                    manageTabIndex
                    typeToSelect
                    value={this.state.value}
                    aria-label="Filter By Signal Types"
                    onSelect={this.handleSignalTypeSelect}>
                    <SideNavItem value="ALL">All</SideNavItem>
                    {categoryTypes.map(categoryType => (
                        <SideNavHeading label={`${categoryType} SIGNALS`} key={categoryType}>
                            {signalTypeOptions
                                .filter(({ category }) => category === categoryType)
                                .map(({ value, label }) => (
                                    <SideNavItem value={value} key={value}>
                                        {label}
                                    </SideNavItem>
                                ))}
                        </SideNavHeading>
                    ))}
                </SideNav>
            </Fragment>
        );
    }
}

SignalTypeFilter.propTypes = {
    isSearched: PropTypes.bool,
    onSignalTypeChange: PropTypes.func,
    signalType: PropTypes.string,
};

export default SignalTypeFilter;
