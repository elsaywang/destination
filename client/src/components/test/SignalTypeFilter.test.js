import React from 'react';
import { shallow } from 'enzyme';
import SignalTypeFilter from '../SignalTypeFilter';
import Heading from '@react/react-spectrum/Heading';
import { Tab, TabList } from '@react/react-spectrum/TabList';
import { getSignalTypeOptions } from '../../constants/signalTypeOptions';

describe('<SignalTypeFilter /> component', () => {
    const mockFn = jest.fn();
    const props = {
        signalType: 'ALL',
    };
    const state = {
        counts: {
            ALL: 72093,
            ANALYTICS: 34300,
            ALF: 359,
            REALTIME: 27,
            ONBOARDED: 37407,
        },
    };
    const wrapper = shallow(
        <SignalTypeFilter onSignalTypeChange={mockFn} signalType={props.signalType} />,
    );
    const signalTypeOptions = getSignalTypeOptions(state.counts);

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Heading /> component', () => {
            expect(wrapper.find(Heading).exists()).toBe(true);
        });

        it('renders <TabList /> component with correct number of <Tab />s', () => {
            expect(wrapper.find(TabList).exists()).toBe(true);
            expect(wrapper.find(Tab).length).toBe(signalTypeOptions.length);
        });

        it('renders correctly selected <Tab /> component when props.signalType change', () => {
            const signalType = 'ANALYTICS';
            const optionMatchingSignalType = signalTypeOptions.find(
                option => option.value === signalType,
            ).label;

            wrapper.setProps({ signalType });

            expect(
                wrapper
                    .find(Tab)
                    .filterWhere(tab => tab.props().selected)
                    .html(),
            ).toContain(optionMatchingSignalType);
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Tab /> component with correct label when props.count change', () => {
            wrapper
                .find(Tab)
                .map((tab, index) => expect(tab.html()).toContain(signalTypeOptions[index].label));

            const newState = {
                counts: {
                    ALL: 1100,
                    ANALYTICS: 1000,
                    ALF: 100,
                    REALTIME: 0,
                    ONBOARDED: 0,
                },
            };
            const newSignalTypeOptions = getSignalTypeOptions(newState.counts);

            wrapper.setState({
                ...newState,
            });

            wrapper
                .find(Tab)
                .map((tab, index) =>
                    expect(tab.html()).toContain(newSignalTypeOptions[index].label),
                );
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('event handlers', () => {
        it('.handleSignalTypeChange() calls `onSignalTypeChange` prop with a signalType that corresponds to index of the signal type tab clicked', () => {
            wrapper.instance().handleSignalTypeChange(0);
            expect(wrapper.instance().props.onSignalTypeChange).toHaveBeenCalledWith('ALL');
            wrapper.instance().handleSignalTypeChange(1);
            expect(wrapper.instance().props.onSignalTypeChange).toHaveBeenCalledWith('ANALYTICS');
            wrapper.instance().handleSignalTypeChange(2);
            expect(wrapper.instance().props.onSignalTypeChange).toHaveBeenCalledWith('ALF');
            wrapper.instance().handleSignalTypeChange(3);
            expect(wrapper.instance().props.onSignalTypeChange).toHaveBeenCalledWith('REALTIME');
            wrapper.instance().handleSignalTypeChange(4);
            expect(wrapper.instance().props.onSignalTypeChange).toHaveBeenCalledWith('ONBOARDED');
        });
    });
});
