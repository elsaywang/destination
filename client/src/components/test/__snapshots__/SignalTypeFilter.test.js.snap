// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignalTypeFilter /> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalTypeFilter
    counts={
      Object {
        "ALF": 359,
        "ALL": 72093,
        "ANALYTICS": 34300,
        "ONBOARDED": 37407,
        "REALTIME": 27,
      }
    }
    onSignalTypeChange={[MockFunction]}
    signalType="ALL"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Heading
          className={undefined}
          size={3}
        >
          Filter by Signal Type
        </Heading>,
        <TabList
          className={undefined}
          onChange={[Function]}
          orientation="vertical"
          quiet={false}
          value="ALL"
          variant=""
        >
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={true}
          >
            All (72093)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Adobe Analytics (34300)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Actionable Log Files (359)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            General Online Data (27)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Onboarded Records (37407)
          </Tab>
        </TabList>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Filter by Signal Type",
          "className": undefined,
          "size": 3,
        },
        "ref": null,
        "rendered": "Filter by Signal Type",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              All (72093)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Adobe Analytics (34300)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>,
          ],
          "className": undefined,
          "onChange": [Function],
          "orientation": "vertical",
          "quiet": false,
          "value": "ALL",
          "variant": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "ALL",
            "nodeType": "class",
            "props": Object {
              "children": "All (72093)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": true,
            },
            "ref": null,
            "rendered": "All (72093)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ANALYTICS",
            "nodeType": "class",
            "props": Object {
              "children": "Adobe Analytics (34300)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Adobe Analytics (34300)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ALF",
            "nodeType": "class",
            "props": Object {
              "children": "Actionable Log Files (359)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Actionable Log Files (359)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "REALTIME",
            "nodeType": "class",
            "props": Object {
              "children": "General Online Data (27)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "General Online Data (27)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ONBOARDED",
            "nodeType": "class",
            "props": Object {
              "children": "Onboarded Records (37407)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Onboarded Records (37407)",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Heading
            className={undefined}
            size={3}
          >
            Filter by Signal Type
          </Heading>,
          <TabList
            className={undefined}
            onChange={[Function]}
            orientation="vertical"
            quiet={false}
            value="ALL"
            variant=""
          >
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              All (72093)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Adobe Analytics (34300)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>
          </TabList>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Filter by Signal Type",
            "className": undefined,
            "size": 3,
          },
          "ref": null,
          "rendered": "Filter by Signal Type",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={true}
              >
                All (72093)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Adobe Analytics (34300)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Actionable Log Files (359)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                General Online Data (27)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Onboarded Records (37407)
              </Tab>,
            ],
            "className": undefined,
            "onChange": [Function],
            "orientation": "vertical",
            "quiet": false,
            "value": "ALL",
            "variant": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "ALL",
              "nodeType": "class",
              "props": Object {
                "children": "All (72093)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": true,
              },
              "ref": null,
              "rendered": "All (72093)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ANALYTICS",
              "nodeType": "class",
              "props": Object {
                "children": "Adobe Analytics (34300)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Adobe Analytics (34300)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ALF",
              "nodeType": "class",
              "props": Object {
                "children": "Actionable Log Files (359)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Actionable Log Files (359)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "REALTIME",
              "nodeType": "class",
              "props": Object {
                "children": "General Online Data (27)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "General Online Data (27)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ONBOARDED",
              "nodeType": "class",
              "props": Object {
                "children": "Onboarded Records (37407)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Onboarded Records (37407)",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SignalTypeFilter /> component rendering renders <Tab /> component with correct label when props.count change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalTypeFilter
    counts={
      Object {
        "ALF": 100,
        "ALL": 1100,
        "ANALYTICS": 1000,
        "ONBOARDED": 0,
        "REALTIME": 0,
      }
    }
    onSignalTypeChange={[MockFunction]}
    signalType="ANALYTICS"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Heading
          className={undefined}
          size={3}
        >
          Filter by Signal Type
        </Heading>,
        <TabList
          className={undefined}
          onChange={[Function]}
          orientation="vertical"
          quiet={false}
          value="ANALYTICS"
          variant=""
        >
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            All (1100)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={true}
          >
            Adobe Analytics (1000)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Actionable Log Files (100)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            General Online Data (0)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Onboarded Records (0)
          </Tab>
        </TabList>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Filter by Signal Type",
          "className": undefined,
          "size": 3,
        },
        "ref": null,
        "rendered": "Filter by Signal Type",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (1100)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (1000)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (100)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (0)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (0)
            </Tab>,
          ],
          "className": undefined,
          "onChange": [Function],
          "orientation": "vertical",
          "quiet": false,
          "value": "ANALYTICS",
          "variant": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "ALL",
            "nodeType": "class",
            "props": Object {
              "children": "All (1100)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "All (1100)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ANALYTICS",
            "nodeType": "class",
            "props": Object {
              "children": "Adobe Analytics (1000)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": true,
            },
            "ref": null,
            "rendered": "Adobe Analytics (1000)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ALF",
            "nodeType": "class",
            "props": Object {
              "children": "Actionable Log Files (100)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Actionable Log Files (100)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "REALTIME",
            "nodeType": "class",
            "props": Object {
              "children": "General Online Data (0)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "General Online Data (0)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ONBOARDED",
            "nodeType": "class",
            "props": Object {
              "children": "Onboarded Records (0)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Onboarded Records (0)",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Heading
            className={undefined}
            size={3}
          >
            Filter by Signal Type
          </Heading>,
          <TabList
            className={undefined}
            onChange={[Function]}
            orientation="vertical"
            quiet={false}
            value="ANALYTICS"
            variant=""
          >
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (1100)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (1000)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (100)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (0)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (0)
            </Tab>
          </TabList>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Filter by Signal Type",
            "className": undefined,
            "size": 3,
          },
          "ref": null,
          "rendered": "Filter by Signal Type",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                All (1100)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={true}
              >
                Adobe Analytics (1000)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Actionable Log Files (100)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                General Online Data (0)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Onboarded Records (0)
              </Tab>,
            ],
            "className": undefined,
            "onChange": [Function],
            "orientation": "vertical",
            "quiet": false,
            "value": "ANALYTICS",
            "variant": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "ALL",
              "nodeType": "class",
              "props": Object {
                "children": "All (1100)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "All (1100)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ANALYTICS",
              "nodeType": "class",
              "props": Object {
                "children": "Adobe Analytics (1000)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": true,
              },
              "ref": null,
              "rendered": "Adobe Analytics (1000)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ALF",
              "nodeType": "class",
              "props": Object {
                "children": "Actionable Log Files (100)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Actionable Log Files (100)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "REALTIME",
              "nodeType": "class",
              "props": Object {
                "children": "General Online Data (0)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "General Online Data (0)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ONBOARDED",
              "nodeType": "class",
              "props": Object {
                "children": "Onboarded Records (0)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Onboarded Records (0)",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SignalTypeFilter /> component rendering renders correctly selected <Tab /> component when props.signalType change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalTypeFilter
    counts={
      Object {
        "ALF": 359,
        "ALL": 72093,
        "ANALYTICS": 34300,
        "ONBOARDED": 37407,
        "REALTIME": 27,
      }
    }
    onSignalTypeChange={[MockFunction]}
    signalType="ANALYTICS"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Heading
          className={undefined}
          size={3}
        >
          Filter by Signal Type
        </Heading>,
        <TabList
          className={undefined}
          onChange={[Function]}
          orientation="vertical"
          quiet={false}
          value="ANALYTICS"
          variant=""
        >
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            All (72093)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={true}
          >
            Adobe Analytics (34300)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Actionable Log Files (359)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            General Online Data (27)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Onboarded Records (37407)
          </Tab>
        </TabList>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Filter by Signal Type",
          "className": undefined,
          "size": 3,
        },
        "ref": null,
        "rendered": "Filter by Signal Type",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (72093)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (34300)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>,
          ],
          "className": undefined,
          "onChange": [Function],
          "orientation": "vertical",
          "quiet": false,
          "value": "ANALYTICS",
          "variant": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "ALL",
            "nodeType": "class",
            "props": Object {
              "children": "All (72093)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "All (72093)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ANALYTICS",
            "nodeType": "class",
            "props": Object {
              "children": "Adobe Analytics (34300)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": true,
            },
            "ref": null,
            "rendered": "Adobe Analytics (34300)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ALF",
            "nodeType": "class",
            "props": Object {
              "children": "Actionable Log Files (359)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Actionable Log Files (359)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "REALTIME",
            "nodeType": "class",
            "props": Object {
              "children": "General Online Data (27)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "General Online Data (27)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ONBOARDED",
            "nodeType": "class",
            "props": Object {
              "children": "Onboarded Records (37407)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Onboarded Records (37407)",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Heading
            className={undefined}
            size={3}
          >
            Filter by Signal Type
          </Heading>,
          <TabList
            className={undefined}
            onChange={[Function]}
            orientation="vertical"
            quiet={false}
            value="ANALYTICS"
            variant=""
          >
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (72093)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (34300)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>
          </TabList>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Filter by Signal Type",
            "className": undefined,
            "size": 3,
          },
          "ref": null,
          "rendered": "Filter by Signal Type",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                All (72093)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={true}
              >
                Adobe Analytics (34300)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Actionable Log Files (359)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                General Online Data (27)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Onboarded Records (37407)
              </Tab>,
            ],
            "className": undefined,
            "onChange": [Function],
            "orientation": "vertical",
            "quiet": false,
            "value": "ANALYTICS",
            "variant": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "ALL",
              "nodeType": "class",
              "props": Object {
                "children": "All (72093)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "All (72093)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ANALYTICS",
              "nodeType": "class",
              "props": Object {
                "children": "Adobe Analytics (34300)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": true,
              },
              "ref": null,
              "rendered": "Adobe Analytics (34300)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ALF",
              "nodeType": "class",
              "props": Object {
                "children": "Actionable Log Files (359)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Actionable Log Files (359)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "REALTIME",
              "nodeType": "class",
              "props": Object {
                "children": "General Online Data (27)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "General Online Data (27)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ONBOARDED",
              "nodeType": "class",
              "props": Object {
                "children": "Onboarded Records (37407)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Onboarded Records (37407)",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
